modules/schema/src/main/java/org/apache/ignite/internal/schema/BinaryTupleComparatorUtils.java

```diff
@@ -122,6 +122,12 @@ static int compareAsString(BinaryTupleReader tuple, int colIndex, String cmp, bo
         byte[] bytes = tuple.bytesValue(begin, begin + trimmedSize);
         char[] cmpArray = cmp.toCharArray();
 
+        if (bytes.length < trimmedSize) {
```

Comment:
Please add a comment that this "if" does and why it is needed

Reply:
Added comment.